File: EjerciciosPython/Ejercicio02/Moto.py
# -*- coding: utf-8 -*-
"""
Created on Tue Jan 10 20:54:37 2023

@author: Usuario
"""

from Vehiculo import *

class Moto(Vehiculo):
    
    def __init__(self, ide, marca, modelo, km, precio, sidecar):
        super().__init__(ide, marca, modelo, km, precio)
        self.__sidecar = sidecar
    
    def getPrecio(self):
        if(self.__sidecar == True):
            return super().getPrecio() + 50
        else:
            return super().getPrecio()
    
    def __str__(self):
        if(self.__sidecar == True):
            return super().__str__() + "\nSi tiene Sidecar"
        else:
            return super().__str__() + "\nNo tiene Sidecar"

File: EjerciciosClasesCSharp/Ejercicio06/Moto.cs
using System;

namespace Ejercicio06
{
    class Moto : Vehiculo
    {
        private bool _sidecar;

        public Moto (string marca, string modelo, int km, int precio, bool sidecar) : base (marca, modelo, km, precio)
        {
            _sidecar = sidecar;
        }

        //Getter y Setter
        public bool sidecar
        {
            get
            {
                return _sidecar;
            }
            set
            {
                _sidecar = value;
            }
        }

        //Modificando metodos padre
        public override int precio 
        { 
            get
            {
                if(sidecar == true)
                {
                    return base.precio += 50;
                }
                else
                {
                    return base.precio;
                }
            }
        }

        public override string ToString()
        {
            if(sidecar == true)
            {
                return base.ToString() + "\nSi tiene un Sidecar";
            }
            else
            {
                return base.ToString() + "\nNo tiene un sidecar";
            }
        }
    }
}